File: worker/tests/fixtures/sast/vulnerable.py
import os
import subprocess
import pickle

# Hardcoded secret — semgrep p/secrets should flag
password = "supersecret123"
api_key = "AKIAIOSFODNN7EXAMPLE"

# Dangerous eval with user input — semgrep p/security-audit
user_input = input("cmd: ")
x = eval(user_input)

# Subprocess with shell and user input — semgrep python.lang.security.audit.dangerous-subprocess
subprocess.run(user_input, shell=True)

# SQL injection with string concat — semgrep
query = "SELECT * FROM users WHERE id = " + user_input

# Unsafe deserialization
data = pickle.loads(b"test")

# Additional file for multi-file proof

File: worker/tests/fixtures/sast/vulnerable.js
// Hardcoded secret
const apiKey = "supersecret123";
const awsKey = "AKIAIOSFODNN7EXAMPLE";

// Dangerous eval with user input
const userInput = process.argv[2];
eval(userInput);

// Function constructor
const f = new Function("a", "b", "return a + b");

// Child process with user input — semgrep javascript.lang.security.audit.dangerous-subprocess
const { exec } = require('child_process');
exec(userInput, (err, stdout) => {});

// SQL injection
const query = "SELECT * FROM users WHERE id = " + userInput;

// XSS via innerHTML with user input
document.body.innerHTML = userInput;
